workspaces/cli-server/src/spectacle/index.ts

```diff
@@ -109,6 +114,40 @@ export class LocalCliSpecRepository implements IOpticSpecReadWriteRepository {
       return events;
     }
   }
+
+  async resetToCommit(batchCommitId: string): Promise<void> {
+    const tempOutputFile = path.join(
+      this.dependencies.specDirPath,
+      `optic-temp-spec-file-${uuidv4()}.json`
+    );
+    const tempFileDeletion = path.join(
+      this.dependencies.specDirPath,
+      `optic-temp-spec-to-delete-${uuidv4()}.json`
+    );
+    const specJsFile = path.join(
+      this.dependencies.specDirPath,
+      'specification.json'
+    );
```

Comment:
It turns out we did need these temp files. If you open a read file stream and try write to that same file, it turns out that doesn't work and the parser just crashes. 

I prefixed the temp files as `optic-temp` and added this to the `.gitignore` file. The problem is older optic installs won't have gitignore. This brings up the question how do we update the local states of users with outdated versions (perhaps we could add an optic version to optic.yml and then run relevant "migration" steps) - this means we could also extract out spec id from the spec.json + add in missing gitignores. Something we'll probably want to look into later.